File: Python/validParenthesis.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 1 or s[0] in ') } ]':
            return False
        
        stack = []
        
        for c in s:
            if c in '({[':
                stack.append(c)
            elif c in ')}]':
                if stack:
                    top = stack[-1]
                    if (top == '(' and c == ')') or (top == '{' and c == '}') or (top == '[' and c == ']'):
                        stack.pop()
                    else:
                        return False
                else:
                    return False
        
        return not stack

File: Java/validParenthesis.java
import java.util.*;
class Solution {
    public boolean isValid(String s) {

        if(s.length()==1 || s.charAt(0)==')' || s.charAt(0)=='}' || s.charAt(0)==']')
        return false;

        Stack<Character> open= new Stack<>();

        for(int i=0; i<s.length(); i++){
            char c= s.charAt(i);
            if(c=='(' || c=='{' || c=='['){
                open.push(c);
            }else if(c==')' || c=='}' || c==']'){
                if(open.size()>0 && open.peek()=='(' && c==')')
                open.pop();
                else if(open.size()>0 && open.peek()=='{' && c=='}')
                open.pop();
                else if(open.size()>0 && open.peek()=='[' && c==']')
                open.pop();
                else
                return false;
            }
        }

        return open.size()==0;
    }
}